lib/puppeteer_environment.js

```diff
@@ -0,0 +1,33 @@
+const chalk = require('chalk')
+const NodeEnvironment = require('jest-environment-node')
+const puppeteer = require('puppeteer')
+const fs = require('fs')
+const os = require('os')
+const path = require('path')
+
+const DIR = path.join(os.tmpdir(), 'jest_puppeteer_global_setup')
+
+class PuppeteerEnvironment extends NodeEnvironment {
+  async setup () {
+    console.log(chalk.yellow('Setup Test Environment.'))
+    await super.setup()
+    const wsEndpoint = fs.readFileSync(path.join(DIR, 'wsEndpoint'), 'utf8')
+    if (!wsEndpoint) {
+      throw new Error('wsEndpoint not found')
+    }
+    this.global.__BROWSER__ = await puppeteer.connect({
+      browserWSEndpoint: wsEndpoint
+    })
+  }
+
+  async teardown () {
+    console.log(chalk.yellow('Teardown Test Environment.'))
+    await super.teardown()
+  }
+
+  runScript (script) {
```

Comment:
Do we need to define this function if all we're doing is calling the parent method? Does it not just inherit it?

Reply:
It's not being used now, but gives the space to allowing extra code before running the script. Can remove.